cmd/dmsg-discovery/commands/root.go

```diff
@@ -96,6 +133,47 @@ func prepareDB(log logrus.FieldLogger) store.Storer {
 	return db
 }
 
+func getServers(ctx context.Context, a *api.API, log logrus.FieldLogger) (servers []*disc.Entry) {
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+	for {
+		servers, err := a.AllServers(ctx, log)
+		if err != nil {
+			log.WithError(err).Fatal("Error getting dmsg-servers.")
+		}
+		if len(servers) > 0 {
+			return servers
+		}
+		log.Warn("No dmsg-servers found, trying again in 1 minute.")
```

Comment:
Interval is set to 10 seconds I think.

Reply:
Would it not be better to have the logic from line 140-157 inside the `ticker.C` case instead of calling the function `getServers` recursively?

Reply:
The 10 seconds was for testing, will change to a minute. 
The `getServers` function is to get the `dmsg-servers` before we serve the dmsg as we cannot serve on other `dmsg-servers` after we have already served. Basically, it will only call itself after one minute only if there are zero `dmsg-servers` received. This can be changed after we add [this](https://github.com/skycoin/dmsg/issues/133) feature.